File: frontend/models/documentModels.ts
// models/documentModels.ts
export enum DocumentStatus {
  PENDING = "PENDING",
  VIEWED = "VIEWED",
  EDITED = "EDITED"
}

export enum DocumentPermission {
  VIEW = "VIEW",
  EDIT = "EDIT",
  REVIEW = "REVIEW"
}

export interface Document {
  id: string;
  title: string;
  owner: string;
  file_path: string;
  created_at: string;
  status: DocumentStatus;
}

export interface SharedDocument {
  document_id: string;
  recipient: string;
  permission: DocumentPermission;
  shared_at: string;
  status: DocumentStatus;
}

File: backend/models/document_models.py
from enum import Enum
import uuid
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class DocumentStatus(str, Enum):
    PENDING = "PENDING"
    VIEWED = "VIEWED"
    EDITED = "EDITED"

class DocumentPermission(str, Enum):
    VIEW = "VIEW"
    EDIT = "EDIT"

class DocumentBase(BaseModel):
    title: str
    owner_username: str
    file_path: str
    file_size: Optional[int] = None
    file_type: Optional[str] = None
    description: Optional[str] = None 

class DocumentCreate(DocumentBase):
    pass

class Document(DocumentBase):
    id: uuid.UUID
    created_at: datetime
    status: DocumentStatus
    
    class Config:
        from_attributes = True

class SharedDocumentBase(BaseModel):
    document_id: uuid.UUID
    recipient_username: str
    permission: DocumentPermission
    file_type: str
    title: str
    owner_username: str

class SharedDocumentCreate(SharedDocumentBase):
    pass

class SharedDocument(SharedDocumentBase):
    shared_at: datetime
    status: DocumentStatus
    
    class Config:
        from_attributes = True